client/my-sites/themes/install-theme-button.jsx

```diff
@@ -6,79 +6,82 @@ import { recordTracksEvent } from 'calypso/state/analytics/actions';
 import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
 import isAtomicSite from 'calypso/state/selectors/is-site-automated-transfer';
 import siteHasFeature from 'calypso/state/selectors/site-has-feature';
-import { isJetpackSiteMultiSite, isJetpackSite } from 'calypso/state/sites/selectors';
-import { getSelectedSiteId, getSelectedSiteSlug } from 'calypso/state/ui/selectors';
+import {
+	isJetpackSiteMultiSite,
+	isJetpackSite,
+	getSiteAdminUrl,
+	getSiteSlug,
+} from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
 import { trackClick } from './helpers';
 
 import './install-theme-button.scss';
 
+function getInstallThemeUrl( state, siteId ) {
+	if ( ! siteId ) {
+		return '/themes/upload';
+	}
+
+	const atomicSite = isAtomicSite( state, siteId );
+	const siteCanInstallThemes = siteHasFeature( state, siteId, FEATURE_INSTALL_THEMES );
+	if ( atomicSite && siteCanInstallThemes ) {
+		return getSiteAdminUrl( state, siteId, 'theme-install.php' );
```

Comment:
This may be a useful selector FWIW - it's used in a few places. Could be useful if we want to centralize the atomic-nonatomic logic too